### ISSUE TITLE
Causal_Forest: systematically underestimate treatment effect when TE is large in the usage example

### ISSUE THREAD

**Issue by amilymoor** [swager/grf#174]
While migrating from the previous -causalTree- package to -grf-, I noticed a curious case for -grf- when running the following causal_forest example. When I expand the range of true treatment effect (changed X.test[,1] to be a sequence between -2 and 4), the estimated Treatment effect is systematically below the truth when TE is >2. Any suggestion why this might happen is hugely appreciated.

Below from the usage example:

library(grf)

## Generate data.
n = 2000; p = 10
X = matrix(rnorm(n*p), n, p)
X.test = matrix(0, 101, p)
X.test[,1] = seq(-2, 4, length.out = 101)  # this is the only line I changed from the example

## Perform treatment effect estimation.
W = rbinom(n, 1, 0.5)
Y = pmax(X[,1], 0) * W + X[,2] + pmin(X[,3], 0) + rnorm(n)
tau.forest = causal_forest(X, Y, W, num.trees = 4000)
tau.hat = predict(tau.forest, X.test, estimate.variance = TRUE)
sigma.hat = sqrt(tau.hat$variance.estimates)
plot(X.test[,1], tau.hat$predictions, ylim = range(tau.hat$predictions + 1.96 * sigma.hat, tau.hat$predictions - 1.96 * sigma.hat, 0, 4), xlab = "x", ylab = "tau", type = "l")
lines(X.test[,1], tau.hat$predictions + 1.96 * sigma.hat, col = 1, lty = 2)
lines(X.test[,1], tau.hat$predictions - 1.96 * sigma.hat, col = 1, lty = 2)
lines(X.test[,1], pmax(0, X.test[,1]), col = 2, lty = 1)

## Can see that the estimate systematically goes below the truth (in red).

<img width="546" alt="screen shot 2018-01-29 at 12 50 36 am" src="https://user-images.githubusercontent.com/16353845/35495339-79f30ea8-048e-11e8-9bad-76ae1944b779.png">

**Comment by swager** [comment]
Thanks for the example. The reason forests sometimes misbehave like this near the edges is that forests rely on individual tree-splits to express heterogeneity. Thus, tree predictive surfaces are "flat" in areas without many splits, and this can manifest itself via tapered off response surfaces near the edges.

@rinafriedberg is actively working on a fix to this issue via local linear corrections; hopefully we'll have an update ready soon!

**Comment by amilymoor** [comment]
Thank you for clarification! Is it fair to say that the same problem would occur to the original causalTree package? Currently, is there a way to quantify the trustworthiness of the estimate (e.g. in the above example, we tend to doubt the tapered off response surface and may decide to drop the result from that area in further analysis)? Thanks!

**Comment by swager** [comment]
Yes -- I believe this problem occurs with any existing RF implementation. As a rule of thumb, random forests (and most other local non-parametric methods also) are biased towards ``flattening out'' signals; in particular, they may miss out on small bumps in the regression surface, or fail to track steep slopes near the edge of feature space. For a discussion of this phenomenon is a very simple case, see Figure 5.9 of ''The Elements of Statistical Learning''.

**Comment by amilymoor** [comment]
Thanks a lot! This really helps.
